akka-stream-tests/src/test/scala/akka/stream/scaladsl/QueueSinkSpec.scala

```diff
@@ -109,7 +110,7 @@ class QueueSinkSpec extends StreamSpec {
       sub.sendComplete()
       Await.result(queue.pull(), noMsgTimeout) should be(None)
 
-      queue.pull().failed.foreach { e ⇒ e.isInstanceOf[IllegalStateException] should ===(true) }
+      queue.pull().failed.futureValue shouldBe an[AbruptStageTerminationException]
```

Comment:
I don't think it _should_ be `AbruptStageTerminationException` though, that should only be used to fail stages when the Materialiser is shutdown while the stream is still running. I think either `IllegalState` as expected before, or a new specific exception for when the queue is completed.

Do you mind looking into that @agolubev ?